agents/ahv/fence_ahv.py

```diff
@@ -0,0 +1,688 @@
+                        raise AHVFenceAgentException("No arguments provided")
+
+                if not vm_uuid:
+                        vm_uuid = self._get_vm_uuid(vm_name)
+
+                if power_state.lower() == 'on':
+                        resp = self._power_on_vm(vm_uuid)
+                elif power_state.lower() == 'off':
+                        resp = self._power_off_vm(vm_uuid)
+
+                task_id = resp.json()['data']['extId']
+
+                try:
+                        self._wait_for_task(task_id, timeout)
+                except AHVFenceAgentException as err:
+                        logging.error("Failed to power %s VM", power_state.lower())
+                        logging.error("VM power %s task failed with status, %s",
+                                      power_state.lower(), status)
+                        raise AHVFenceAgentException from err
+
+                logging.info("Powered %s VM, %s successfully",
+                             power_state.lower(), vm_name)
+
+        def power_cycle_vm(self, vm_name=None, vm_uuid=None, timeout=None):
+                """
+                Power cycle a VM
+                """
+                resp = None
+                status = None
+
+                if not timeout:
+                        timeout = MIN_TIMEOUT
+
+                if not vm_name and not vm_uuid:
+                        logging.error("Require at least one of VM name or VM UUID")
+                        raise AHVFenceAgentException("No arguments provided")
+
+                if not vm_uuid:
+                        vm_uuid = self._get_vm_uuid(vm_name)
+
+                resp = self._power_cycle_vm(vm_uuid)
+                task_id = resp.json()['data']['extId']
+
+                try:
+                        self._wait_for_task(task_id, timeout)
+                except AHVFenceAgentException as err:
+                        logging.error("Failed to power-cycle VM %s", vm_uuid)
+                        logging.error("VM power-cycle task failed with status, %s", status)
+                        raise AHVFenceAgentException from err
+
+                logging.info("Power-cycled VM, %s", vm_name)
+
+
+def connect(options):
+        """
+        Create a client instance and return it after verifying
+        that the client can connect.
+        """
+        host = options["--ip"]
+        username = options["--username"]
+        password = options["--password"]
+        verify_ssl = False
+
+        if "--ssl-secure" in options:
+                verify_ssl = True
+
+        client = V4NutanixClient(host, username, password, verify_ssl)
+
+        try:
+                client.list_vms(limit=1)
+        except AHVFenceAgentException as err:
+                logging.error("Connection to Prism Central Failed")
+                logging.error(err)
+                fencing.fail(fencing.EC_LOGIN_DENIED)
+
+        return client
+
+def list_vms(client, options):
```

Comment:
This name is confusing, if fence gent requires to have this name, then please change the name of the function under the class!

Reply:
Done. Changed the outer function to get_list()